### PR TITLE
[MM-54820] Convert ./components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx from Class Component to Function Component

### PR DESCRIPTION
**Pull request by sp6370**
#### Summary
This pull request migrates the existing unmute_channel_button component to a react functional component.

#### Ticket Link
Fixes: https://github.com/mattermost/mattermost/issues/24781
Jira: https://mattermost.atlassian.net/browse/MM-54820

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx [hunk 1]
```diff
@@ -9,36 +9,28 @@ import {NotificationLevels} from 'utils/constants';
...
+            type='button'
+            className='navbar-toggle icon icon__mute'
+            onClick={handleClick}
+        >
+            <span className='fa fa-bell-slash-o icon'/>
+        </button>
+    );
+};
+
+export default UnmuteChannelButton;
```
Comment (larkox) [inline webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx:36]: It used to be a pure component, so use here a `React.memo`.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Just one minor detail.

File: webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx (hunk 1 above)
Comment (sp6370) [reply webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx:36]: I have introduced memo for the component.

**Review by larkox — CHANGES_REQUESTED** [review]
Thanks! Two more minor details

File: webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.test.tsx [hunk 2]
```diff
@@ -31,7 +31,7 @@ describe('components/ChannelHeaderMobile/UnmuteChannelButton', () => {
         const props = baseProps;
         props.actions.updateChannelNotifyProps = jest.fn();
 
-        const wrapper = shallow<UnmuteChannelButton>(<UnmuteChannelButton {...props}/>);
+        const wrapper = shallow<typeof UnmuteChannelButton>(<UnmuteChannelButton {...props}/>);
```
[the remark this answers is not in the document]
Comment (sp6370) [reply webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.test.tsx:34]: Resolved!

File: webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx (hunk 1 above)
Comment (sp6370) [reply webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx]: Done!

**Review by larkox — APPROVED** [review]
LGTM. Thanks!

File: webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx [hunk 3]
```diff
@@ -9,36 +9,28 @@ import {NotificationLevels} from 'utils/constants';
...
-                updateChannelNotifyProps,
-            },
-        } = this.props;
-
-        updateChannelNotifyProps(user.id, channel.id, {mark_unread: NotificationLevels.ALL} as ChannelNotifyProps);
+};
+
+const UnmuteChannelButton = ({user, channel, actions}: Props) => {
+    const handleClick = () => {
+        actions.updateChannelNotifyProps(user.id, channel.id, {mark_unread: NotificationLevels.ALL} as ChannelNotifyProps);
```
Comment (sinansonmez) [inline webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx:22]: it is not really in the scope of this PR but can we get rid of typecasting `as ChannelNotifyProps`. 
You can change the `Actions` type from 
```ts
type Actions = {
     updateChannelNotifyProps: (userId: string, channelId: string, props: ChannelNotifyProps) => void;
 };
```
To:

```ts
type Actions = {
     updateChannelNotifyProps: (userId: string, channelId: string, props: Pick<ChannelNotifyProps, 'mark_unread'>) => void;
 };
```

**Review by sinansonmez — CHANGES_REQUESTED** [review]
overall changes are fine but I have minor comment. In addition to comment, there is a failing test (`menu click handlers › should be able to open a React Bootstrap modal from a submenu with the mouse`), could you please check?

File: webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx (hunk 3 above)
Comment (sp6370) [reply webapp/channels/src/components/channel_header_mobile/unmute_channel_button/unmute_channel_button.tsx]: Done!

**Comment by sp6370** [comment]
Hi @sinansonmez!

I am not sure what's exactly wrong with the failing test.

Test which failed:
image

Local Test execution output: 
image

So the specific test passes locally on the most recent commit. Could this be a flaky test?

**Comment by larkox** [comment]
@sp6370 That looks like a flaky test, yes. You can ignore it for now.

**Review by sinansonmez — APPROVED** [review]
LGTM, thanks for the changes :rocket:

**Review by lindy65 — APPROVED** [review]
Thanks @sp6370 - looks good 👍 

cc @larkox

**Comment by Anudhyan** [comment]
I want to test QA

**Comment by lindy65** [comment]
> @lindy65 I want to test QA

Hello @Anudhyan - QA and dev reviews on this PR are already complete (check the labels displayed top right)

**Pull request merged**